app/src/pages/api/report/create.ts

```diff
@@ -0,0 +1,78 @@
+import {
+    BAD_REQUEST,
+    HttpMethod,
+    IApiErrorResponse,
+    IReport,
+    INTERNAL_SERVER_ERROR,
+    METHOD_NOT_ALLOWED,
+} from '@/models';
+import { ReportDbClient, isUndefined } from '@/utils';
+import type { NextApiRequest, NextApiResponse } from 'next';
+
+export interface ICreateReportResponse {
+    report: IReport;
+}
+
+/**
+ * Create new report
+ *
+ * Allowed methods: POST
+ *
+ * Parameters:
+ *  - body: IReport, new report to pUT
+ *
+ * Response: IGetReportResponse
+ *
+ * Potential errors:
+ *  - 400: when body is invalid or not provided
+ *  - 405: when non-allowed method is used
+ *  - 500: internal server error
+ */
+export default async function handler(
+    req: NextApiRequest,
+    res: NextApiResponse<ICreateReportResponse | IApiErrorResponse>
+) {
+    if (req.method !== HttpMethod.POST) {
+        return res.status(405).send({ report: METHOD_NOT_ALLOWED });
+    }
+
+    const { id } = req.query;
```

Comment:
Why are we getting this value? Shouldn't the ID for a new report always be a randomly-generated UUID?